src/map/mob.cpp

```diff
@@ -2403,14 +2400,14 @@ TIMER_FUNC(mob_respawn){
 	return 1;
 }
 
-void mob_log_damage(struct mob_data *md, struct block_list *src, int32 damage)
+void mob_log_damage(struct mob_data *md, struct block_list *src, int32 damage, int32 attdamage)
 {
 	uint32 char_id = 0;
 	int32 flag = MDLF_NORMAL;
 
 	if( damage < 0 )
 		return; //Do nothing for absorbed damage.
-	if( !damage && !(src->type&DEFAULT_ENEMY_TYPE(md)) )
+	if( !damage && !attdamage && !(src->type&DEFAULT_ENEMY_TYPE(md)) )
```

Comment:
```suggestion
	if( ( damage != 0 || attdamage != 0 ) && !(src->type&DEFAULT_ENEMY_TYPE(md)) )
```
Wouldnt that be the more correct logic now?

Reply:
Only if both are 0, you don't want to record it.